crates/core/src/operations/clone.rs

```diff
@@ -0,0 +1,362 @@
+        .try_collect()
+        .await?;
+
+    let mut actions = Vec::with_capacity(file_views.len() + 1);
+    actions.push(Action::Metadata(src_metadata));
+
+    // 4) Add files to the target table
+    let target_root_path = target_table
+        .log_store()
+        .config()
+        .location
+        .to_file_path()
+        .map_err(|_| {
+            DeltaTableError::InvalidTableLocation(format!(
+                "Failed to convert target URL to file path: {}",
+                target.as_ref()
+            ))
+        })?;
+
+    let source_root_path = src_log.config().location.to_file_path().map_err(|_| {
+        DeltaTableError::InvalidTableLocation(format!(
+            "Failed to convert source URL to file path: {}",
+            src_url.as_ref()
+        ))
+    })?;
+
+    for view in file_views {
+        let mut add = view.add_action();
+        add.data_change = true;
+        // Absolute paths are not supported for now, create symlinks instead.
+        add_symlink(
+            source_root_path.as_path(),
+            target_root_path.as_path(),
+            &add.path,
+        )?;
+        actions.push(Action::Add(add));
+    }
+
+    // 5) Commit ADD operations
+    let operation = DeltaOperation::Write {
+        mode: SaveMode::Append,
+        partition_by: if partition_columns.is_empty() {
+            None
+        } else {
+            Some(partition_columns)
+        },
+        predicate: None,
+    };
+
+    let target_snapshot = target_table.snapshot()?.snapshot().clone();
+    let prepared_commit = CommitBuilder::default()
+        .with_actions(actions)
+        .build(Some(&target_snapshot), target_table.log_store(), operation)
+        .into_prepared_commit_future()
+        .await?;
+
+    let log_store = target_table.log_store();
+    let commit_version = target_snapshot.version() + 1;
+    let commit_bytes = prepared_commit.commit_or_bytes();
+    let operation_id = uuid::Uuid::new_v4();
+
+    log_store
+        .write_commit_entry(commit_version, commit_bytes.clone(), operation_id)
+        .await?;
+
+    target_table.update().await?;
+    Ok(target_table)
+}
+
+fn add_symlink(source_root: &Path, target_root: &Path, add_filename: &str) -> std::io::Result<()> {
+    let file_name = Path::new(add_filename);
+    let src_path = source_root.join(file_name);
+    let link_path = target_root.join(file_name);
+
+    // Create parent directories if needed
+    if let Some(parent) = link_path.parent() {
+        std::fs::create_dir_all(parent)?;
+    }
+
+    // Best-effort symlink creation: only when both source and target are local filesystems.
+    #[cfg(target_family = "windows")]
+    {
+        use std::os::windows::fs::symlink_file;
+        symlink_file(&src_path, &link_path)?;
+    }
+    #[cfg(target_family = "unix")]
+    {
+        use std::os::unix::fs::symlink;
+        symlink(&src_path, &link_path)?;
+    }
+    Ok(())
+}
+
+#[cfg(all(test, feature = "datafusion"))]
```

Comment:
I ended up requiring datafusion for the test here so that I could verify the data in the clone matched the data in the original at the same version. Not sure if this is a problem.